pkg/kubelet/qos/policy_test.go

```diff
@@ -159,81 +171,270 @@ var (
+			lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
+				"burstable-unique-container": {lowOOMScoreAdj: 875, highOOMScoreAdj: 880},
+			},
+			sidecarContainersFeatureEnabled: true,
 		},
 		{
-			pod:             &nodeCritical,
-			memoryCapacity:  4000000000,
-			lowOOMScoreAdj:  -997,
-			highOOMScoreAdj: -997,
+			pod:            &burstableMixedUniqueMainContainerPod,
+			memoryCapacity: standardMemoryAmount,
+			lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
+				"init-container": {lowOOMScoreAdj: 875, highOOMScoreAdj: 880},
+				"main-1":         {lowOOMScoreAdj: 875, highOOMScoreAdj: 880},
+			},
+			sidecarContainersFeatureEnabled: true,
+		},
+		{
+			pod:            &burstableMixedMultiContainerSmallSideCarePod,
+			memoryCapacity: standardMemoryAmount,
+			lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
+				"init-container":          {lowOOMScoreAdj: 875, highOOMScoreAdj: 880},
+				"sidecar-small-container": {lowOOMScoreAdj: 951, highOOMScoreAdj: 951}, // first to kill
+				"main-1":                  {lowOOMScoreAdj: 875, highOOMScoreAdj: 875},
+			},
+			sidecarContainersFeatureEnabled: true,
+		},
+		{
+			pod:            &burstableMixedMultiContainerSameRequestPod,
+			memoryCapacity: standardMemoryAmount,
+			lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
+				"init-container":    {lowOOMScoreAdj: 875, highOOMScoreAdj: 880},
+				"sidecar-container": {lowOOMScoreAdj: 876, highOOMScoreAdj: 876}, // first to kill
+				"main-1":            {lowOOMScoreAdj: 875, highOOMScoreAdj: 875},
+			},
+			sidecarContainersFeatureEnabled: true,
+		},
+		{
+			pod:            &burstableMixedMultiContainerBigSidecarContainerPod,
+			memoryCapacity: standardMemoryAmount,
+			lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
+				"init-container":        {lowOOMScoreAdj: 875, highOOMScoreAdj: 880},
+				"sidecar-big-container": {lowOOMScoreAdj: 876, highOOMScoreAdj: 876}, // first to kill
+				"main-1":                {lowOOMScoreAdj: 875, highOOMScoreAdj: 875},
+			},
+			sidecarContainersFeatureEnabled: true,
 		},
 	}
 	for _, test := range oomTests {
-		oomScoreAdj := GetContainerOOMScoreAdjust(test.pod, &test.pod.Spec.Containers[0], test.memoryCapacity)
-		if oomScoreAdj < test.lowOOMScoreAdj || oomScoreAdj > test.highOOMScoreAdj {
-			t.Errorf("oom_score_adj should be between %d and %d, but was %d", test.lowOOMScoreAdj, test.highOOMScoreAdj, oomScoreAdj)
+		featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SidecarContainers, test.sidecarContainersFeatureEnabled)
+		listContainers := test.pod.Spec.InitContainers
+		listContainers = append(listContainers, test.pod.Spec.Containers...)
+		for _, container := range listContainers {
+			oomScoreAdj := GetContainerOOMScoreAdjust(test.pod, &container, test.memoryCapacity)
+			if oomScoreAdj < test.lowHighOOMScoreAdj[container.Name].lowOOMScoreAdj || oomScoreAdj > test.lowHighOOMScoreAdj[container.Name].highOOMScoreAdj {
```

Comment:
This might be a silly question, but why do you allow specifying both a low and high oomScoreAdj for each container in the test? I noticed that you're checking if the oomScoreAdj calculated for each container falls between the low and high values, but I'm curious why you're not checking if the value itself matches exactly.

Reply:
I agree, but I initially didn’t want to modify the existing logic.

Reply:
I see, I'm glad to know my understanding is correct :)
I'll leave it up to you whether to make changes in this PR.